viper.go

```diff
@@ -1172,7 +1175,9 @@ func (v *Viper) cachedFind(lcaseKey string, flagDefault bool) interface{} {
 	value = v.find(lcaseKey, flagDefault)
 
 	v.lock.Lock()
-	v.cache.Set(realKey, value, 0)
+	if !v.cache.Set(realKey, value, 0) {
+		panic("could not set in cache")
+	}
```

Comment:
```suggestion
	v.cache.Set(realKey, value, 0)
```

Reply:
Shouldn't we check whether it was successfully set? (sure not with a panic)

Reply:
There is no need to, and we also do not have a mitigation strategy except maybe retrying until it passes, which isn't great I think